src/reducers/book.js

```diff
@@ -0,0 +1,32 @@
+const books = [
+  {
+    id: 1,
+    title: 'El Zorro Fuerte',
+    category: 'Action',
+  },
+  {
+    id: 2,
+    title: 'Stephen King Book',
+    category: 'Horror',
+  },
+  {
+    id: 3,
+    title: 'Turkish history for dummies',
+    category: 'History',
+  },
+];
+
+let id = 4;
+
+export default (state = books, action) => {
+  switch (action.type) {
+    case 'ADD_BOOK':
+      const book = action.book;
+      book.id = id++;
```

Comment:
The generation of the ID is better placed in the action (addBook). Why? Ideally the book ID is automatically generated from a backend database and doing it here in the reducer will add another step should you later on decide to extend this. Anyway the instruction implied it should be here.